haskell-src/Concordium/Types.hs

```diff
@@ -507,6 +508,15 @@ newtype VoterPower = VoterPower AmountUnit
 newtype AccountIndex = AccountIndex Word64
     deriving (Eq, Ord, Num, Enum, Bounded, Real, Hashable, Read, Show, Integral, FromJSON, ToJSON, Bits) via Word64
 
+-- |The identifier associated with an account.
+data AccountIdentifier =
+  -- |Given credential registration id as an identifier.
+  CID !CredentialRegistrationID
+  -- |Given address as an identifier. Multiple addresses may refer to the same account.
+  | AA !AccountAddress
+  -- |Given index as an identifier.
+  | AI !AccountIndex
```

Comment:
I think constructors for global types should have longer and more specific names to avoid polluting the namespace.

Reply:
Good catch @td202 , I was already on and pushed the change.